nixos/modules/services/web-servers/caddy/default.nix

```diff
@@ -340,7 +340,7 @@ in
       groups = config.users.groups;
     }) acmeHosts;
 
-    services.caddy.extraConfig = concatMapStringsSep "\n" mkVHostConf virtualHosts;
+    services.caddy.extraConfig = lib.mkAfter (concatMapStringsSep "\n" mkVHostConf virtualHosts);
```

Comment:
What if users use `mkAfter` in their `extraConfig`?

Is this better?
```nix
Caddyfile = pkgs.writeTextDir "Caddyfile" ''
          {
            ${cfg.globalConfig}
          }
          ${cfg.extraConfig}
          ${concatMapStringsSep "\n" mkVHostConf virtualHosts}
        '';
```

Reply:
They could always use `lib.mkOrder` as an alternative, I guess, but I like your idea much more.
Thanks!